doc/HowToPipeline.md

```diff
@@ -0,0 +1,2040 @@
+                                                   dimensions=("visit", "detector", "abstract_filter",
+                                                              "skymap"),
+                                                   storageClass="SourceCatalog",
+                                                   name="src",
+                                                   multiple=True,
+                                                   deferLoad=True) 
+    backgrounds = pipeBase.connectionTypes.Input(doc="Background model for the exposure",
+                                                 storageClass="Background",
+                                                 dimensions=("visit", "detector", "abstract_filter",
+                                                             "skymap"),
+                                                 name="calexpBackground",
+                                                 multiple=True,
+                                                 deferLoad=True)
+    ...
+```
+
+Now Let's see how the `run` method changes to make use of this.
+
+
+```python
+class ApertureTask(pipeBase.PipelineTask):
+
+    ...
+
+    def run(self, exposures: typing.List[afwImage.Exposure],
+            inputCatalogs: typing.List[afwTable.SourceCatalog],
+            areaMasks: typing.List[afwImage.Mask],
+            backgrounds: typing.Union[None, typing.List[afwMath.BackgroundList]] = None,) -> pipeBase.Struct:
+        catalogLengths = []
+        # Add in all the input catalogs into the output catalog
+        for inCatHandle in inputCatalogs:
+            inCat = inCatHandle.get()
+            catalogLengths.append(len(inCat))
+            self.outputCatalog.extend(inCat, mapper=self.mapper)
+
+        # Track which input catalog a given source is coming from
+        inputCatCount = 0
+        cumLength = 0
+
+        # Create some empty variables
+        exposure, areaMask, dimensions, indy, indx = [None]*5
+
+        # This function is an inner function (closure) and has access to variables defined outside
+        # its scope. Based on what input catalog is being processed, it updates each of these
+        # non local variables.
+        def updateVars():
+            nonlocal exposure, areaMask, dimensions, indy, indx, cumLength, inputCatCount
+            exposure = exposure[inputCatCount].get()
+            areaMask = areaMasks[inputCatCount].get()
+
+            # If a background is supplied, add it back to the image so local background subtraction
+            # can be done.
+            if backgrounds is not none:
+                exposure.image.array += backgrounds[inputCatCount].get().getImage()
+
+
+            # Get the dimensions of the exposure
+            dimensions = exposure.getDimensions()
+
+            # Get indexes for each pixel
+            indy, indx = np.indices((dimensions.getY(), dimensions.getX()))
+
+            cumLength += catalogLengths[inputCatCount]
+  
```

Comment:
There's been much debate about what "activator" means in the various middleware working groups and since, without much resolution.  If you can just avoid using the term instead of referring vaguely to "middleware", that'd be preferable, as it's virtually certain any definition of "activator" will conflict with someone else's definition somewhere else.